Use the configuration port when running the app

sirreal opened the pull request:
Various config files set a `port`, or `false` in the case of the shared config:

https://github.com/Automattic/wp-calypso/blob/641207b6f599e9c4e4a11b89821248467b204f4c/config/development.json#L8
https://github.com/Automattic/wp-calypso/blob/641207b6f599e9c4e4a11b89821248467b204f4c/config/test.json#L6
https://github.com/Automattic/wp-calypso/blob/641207b6f599e9c4e4a11b89821248467b204f4c/config/_shared.json#L32

These properties do not seem to affect the port which is used. This PR adds the the config `port` so that it can be configured.

## Testing
* Try adding port via environment, i.e. `PORT=3001 npm start`. This should have precedence.
* Try modifying the port in config files, which should override the default `3000`. This is the behavior corrected by this PR.
* When a config file does not specify a port, the default `3000` should continue to be used.
* Test different environments like `wp-calypso` via the docker build.

**UPDATE**
* Try using oauth to ensure the correct port is used (9aed88f)
* Try removing the port from the environment config file, and ensure the default `3000` is used from `_shared`.

samouri commented:
Whats the signifcance of the `port: false` in `wp-calypso/config/_shared.json`?

samouri commented:
Thanks for this fix @sirreal!

Can you also change the `chalk` calls here: https://github.com/Automattic/wp-calypso/blob/master/server/bundler/index.js#L45

As is, even with a changed port, we will be outputting a:
"Ready! You can load http://calypso.localhost:3000/ now. Have fun!"
to the console.

sirreal commented:
> Whats the significance of the port: `false` in `wp-calypso/config/_shared.json`?

It looks like #11698 was an attempt to gain some consistency across environment configurations which intruduced a `null` value.
Default value was then changed to `false` via #12332.

If this documentation is correct, these are just default values which behave nicely with this change:

https://github.com/Automattic/wp-calypso/blob/6657f23180adfc0dfdf9ebf42dfc10104ab120cc/client/lib/create-config/index.js#L17-L44

sirreal commented:
Thanks both for the input!

I've added the port for oauth in 9aed88f and added it to testing instructions.

The messaging is also fixed in d9d6d54:

<img width="749" alt="after" src="https://user-images.githubusercontent.com/841763/30499764-aaa94a50-9a10-11e7-9c47-a76e6cab36f0.png">

In d4508d1, I moved the default out of `index.js` and into the `_shared` config which should be more correct and prevent needing to write `env || config  || 3000` when we need this value. Added testing instructions to remove the environment value and ensure the default `3000` is used.

sirreal commented on config/_shared.json:
```diff
@@ -29,7 +29,7 @@
 	"oauth_client_id": false,
 	"olark": {},
-	"port": false,
+	"port": 3000,
```
Just leaving a note here. I've been tempted to make this line `process.env.PORT || 3000`. That's dangerous as the environment variable would _only_ override in the case that no env-specific port is defined.

This kind of change could, and perhaps should, happen in the config file. That would allow removing the conditions `env || config` that appear in several places.
